openvdb/points/PointConversion.h

```diff
@@ -773,19 +669,70 @@ createPointDataGrid(const PointIndexGridT& pointIndexGrid, const PositionArrayT&
 
     // create point attribute storage on each leaf
 
-    InitialiseAttributesOp<PointDataTreeT, PointIndexTreeT> initialise(
-                                pointIndexGrid.tree(), descriptor);
-    tbb::parallel_for(leafRange, initialise);
+    leafManager.foreach(
+        [&](LeafT& leaf, size_t /*idx*/) {
+
+            // obtain the PointIndexLeafNode (using the origin of the current leaf)
+            const auto* pointIndexLeaf = pointIndexTree.probeConstLeaf(leaf.origin());
+            assert(pointIndexLeaf);
+
+            // initialise the attribute storage
+            Index pointCount(static_cast<Index>(pointIndexLeaf->indices().size()));
+            leaf.initializeAttributes(descriptor, pointCount);
+        },
+    /*threaded=*/true);
 
     // populate position attribute
 
-    PopulatePositionAttributeOp<PointDataTreeT,
-                                PointIndexTreeT,
-                                PositionArrayT> populate(pointIndexTree,
-                                                        xform,
-                                                        positions);
+    const size_t positionIndex = descriptor->find("P");
+    assert(positionIndex != AttributeSet::INVALID_POS);
+
+    leafManager.foreach(
+        [&](LeafT& leaf, size_t /*idx*/) {
```

Comment:
Is it even necessary to have these as different operators anymore? Can this be merged into the above?

Reply:
Ooh, good suggestion, I hadn't considered that. Just pushed up a change to address this, thanks!

Reply:
Looks good!